defi-pulse/src/adapter.ts

```diff
@@ -0,0 +1,74 @@
+import { Requester, Validator } from '@chainlink/external-adapter'
+import { Execute } from '@chainlink/types'
+import { utils } from 'ethers'
+import { getSymbol } from './symbols'
+import { getAllocations } from './index-allocations'
+import { getPriceAdapter } from './priceAdapter'
+import Decimal from 'decimal.js'
+
+export type Index = IndexAsset[]
+
+export type IndexAsset = {
+  asset: string
+  units: Decimal
+  weight: number
+  price: number
+}
+
+async function getIndex(components: string[], units: number[]): Promise<Index> {
+  const index = []
+  for (let i = 0; i < components.length; i++) {
+    const indexAsset: IndexAsset = {
+      asset: await getSymbol(components[i]),
+      units: new Decimal(new utils.BigNumber(units[i]).toString()).div(1e18),
+      weight: 0,
+      price: 0,
+    }
+    index.push(indexAsset)
+  }
+  return index
+}
+
+const calculateIndex = (index: Index): number =>
```

Comment:
We are calculating the total value here:

```suggestion
const calculateIndexValue = (index: Index): number =>
```